app/src/main/java/com/example/drawingactivity/DrawableAreaView.kt

```diff
@@ -0,0 +1,81 @@
+package com.example.drawingactivity
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.*
+import android.util.AttributeSet
+import android.view.MotionEvent
+import android.view.View
+
+@SuppressLint("ClickableViewAccessibility")
+class DrawableAreaView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0
+) : View(context, attrs, defStyle) {
+
+    var onBitmapDrawn: ((Bitmap) -> Unit)? = null
+    var bitmap: Bitmap? = null
+    private var shape: Rect? = null
+
+    var pathColor = Color.BLACK
+    private var pathPaint = Paint().apply {
+        color = pathColor
+        style = Paint.Style.STROKE
+        strokeWidth = 20f
+        strokeJoin = Paint.Join.ROUND
+        strokeCap = Paint.Cap.ROUND
+    }
+
+    private var bitmapPaint = Paint()
+    lateinit var path: Path
+
+    init {
+        setOnTouchListener { _, mEvent ->
+            when (mEvent.action) {
+                MotionEvent.ACTION_DOWN -> {
+                    pathPaint.color = pathColor
+                    path = Path()
+                    path.moveTo(mEvent.x, mEvent.y)
+                    invalidate()
+                }
+                MotionEvent.ACTION_MOVE -> {
+                    path.lineTo(mEvent.x, mEvent.y)
+                    invalidate()
+                }
+                MotionEvent.ACTION_UP, MotionEvent.ACTION_CANCEL -> {
+                    path.lineTo(mEvent.x, mEvent.y)
+                    bitmap?.let {
+                        onBitmapDrawn?.invoke(it)
+                    }
+                    invalidate()
+                }
+            }
+            true
+        }
+    }
+
+    override fun onDraw(canvas: Canvas?) {
+        super.onDraw(canvas)
+        canvas?.apply {
+            drawColor(Color.LTGRAY)
+            if (bitmap == null) {
+                bitmap = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888)
+                shape = Rect(0, 0, width, height)
+            }
+            bitmap?.let { bitmap ->
+                shape?.let { shape ->
+                    drawBitmap(
+                        bitmap,
+                        null,
+                        shape,
+                        bitmapPaint
```

Comment:
Drawing a bitmap doesn't need a paint. Why not null?

Reply:
Removed the paint and adjusted to null